src/main/java/com/kaloyan/notetakingapp/service/AdminServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.kaloyan.notetakingapp.service;
+
+import com.kaloyan.notetakingapp.dto.UserDTO;
+import com.kaloyan.notetakingapp.model.Role;
+import com.kaloyan.notetakingapp.repository.NoteRepository;
+import com.kaloyan.notetakingapp.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+
+@Service
+public class AdminServiceImpl implements AdminService {
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    NoteRepository noteRepository;
+
+    @Override
+    public Flux<Void> deleteUser(UUID userId) {
+        return Flux.merge(noteRepository.deleteByUserId(userId), userRepository.deleteById(userId));
```

Comment:
this can be automatically handled if the relationship between user and note is configured with CascadeType.REMOVE
what will happen if the user deletion fails? will the notes deletion still get triggered?

Reply:
Spring R2DBC sadly does not support joins including CascadeType